scripts/fmt.sh

```diff
@@ -13,8 +13,18 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.#
 
+# Fix: 'fatal: detected dubious ownership in repository',
+# only do this if it is running into the develop image
+if [ "${PWD}" == "/opt/app" ]; then
+  git config --global --add safe.directory "*"
+fi
+
 pre-commit run --all-files
 RETURN_CODE=$?
+## cat this file for helping on identifying the root cause when some issue happens
+if [ -f /$USER/.cache/pre-commit/pre-commit.log ]; then
```

Comment:
good idea to print the actual errors, but I would prefer to keep the Makefile working consistently on Mac and Linux, but as the  `/$USER/` won't likely exist on Mac, the pre-commit errors will not show up there

Reply:
Not really, it is executed inside the container image, so $USER should point to the correct place.

Reply:
I have `pre-commit` installed on my local system and I run `make fmt` on all the modelmesh projects, without having to spin up a Docker container, or Docker engine.

On a resource restricted machine it makes sense to not require unnecessary overhead.

I think all the messages can be printed to the console with the `-verbose` flag. If we want to only see the output on error, we could cat the log file as you did, but we should not assume a file location.

If we want to use the pre-commit log file, then we should use configure it like described here:
https://pre-commit.com/#config-log_file

i.e. 

```diff
diff --git a/.pre-commit-config.yaml b/.pre-commit-config.yaml
index d7df237..f09997e 100644
--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -20,6 +20,7 @@ repos:
         exclude: ^(gen/)
         args:
           - --skip-dirs='^(gen/)'
+        log_file: .pre-commit.log
   - repo: https://github.com/pre-commit/mirrors-prettier
     rev: v2.4.1
     hooks:
@@ -27,3 +28,4 @@ repos:
         exclude: ^(.github/)
         args:
           - --no-bracket-spacing
+        log_file: .pre-commit.log

```

Reply:
I see, sounds good to me.

But, I've added that check only caring about the CI output since on our machine we can easily change to debug or catch the output, etc.

What about change to `/tmp/pre-commit.log`?

Reply:
On the other PR I changed to `/tmp/pre-commit.log`, let me know if that sounds good.

Reply:
I would create the file in the project and add it to the `.gitignore` and `.dockerignore` files, so when working from the IDE you don't have to traverse directories outside of the project

Reply:
Got it, will apply this here and on runtime-adapter tomorrow morning :)

Thanks!

Reply:
updated.